test/e2e/annotations/rewrite.go

```diff
@@ -195,7 +152,7 @@ var _ = framework.IngressNginxDescribe("Annotations - Rewrite", func() {
 
 		err = f.WaitForNginxServer(host,
 			func(server string) bool {
-				return strings.Contains(server, `location ~* "^/foo" {`) && strings.Contains(server, `location ~* "^/foo.+\/?(?<baseuri>.*)" {`)
+				return strings.Contains(server, `location ~* "^/foo" {`) && strings.Contains(server, `location ~* "^/foo.+" {`)
```

Comment:
Is `location ~* "^/foo" {` not redundant? `location ~* "^/foo.+" {` matches that case as well

Reply:
ah I see it's because of the first ingress